### PR TITLE
Extract TestBorder into a utility file

### PR DESCRIPTION
**Pull request by mehmetf**

### CODE REVIEW

**Comment by mehmetf** [comment]
@Hixie Needs a quick review. The main thing missing is documentation for TestBorder. If you give me a snippet of what it does, I can put it in.

**Review by gspencergoog — APPROVED** [review]

File: packages/flutter/test/widgets/test_border.dart [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import 'package:flutter/material.dart';
+
+typedef Logger = void Function(String caller);
+
```
Comment (gspencergoog) [inline packages/flutter/test/widgets/test_border.dart]: You could add "Test border that logs path accesses for testing purposes." as a description.

**Comment by jonahwilliams** [comment]
We have a number of testing utils besides this one. I'm concerned that by moving this into widgets, we're going to create an incentive to import srcs, or otherwise be unable to change it.

**Comment by mehmetf** [comment]
@jonahwilliams  This is not much better than what's already there but it is also not worse. I am not moving this under lib/widgets, it is going under test/widgets.

Another option I considered is to create another package for this but then it gets complicated since it has to depend on flutter package.

Finally, we could create a separate directory like flutter/test/utils/ but I think that only marginally improves the situation. Did you have something else in mind?

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**